### PR TITLE
first version of cutscene engine

### PR DESCRIPTION
**Pull request by Telvy**

### CODE REVIEW

**Comment by Telvy** [comment]
This is the first version that I managed to commit. Is not exactly what I wanted since most of the system is all show but little substance in the code. I may redo the whole system, but keep certain visual elements like the scrolling text and next icon animation. Thoughts?

File: cutscene_engine/DialogueBox.gd [hunk 1]
```diff
@@ -0,0 +1,54 @@
+extends Control
+
+#export (String, FILE, "*.json") var dialogue_file_path : String
+
+#func interact() -> void:
+	#var dialogue : Dictionary = load_dialogue(dialogue_file_path)
```
Comment (rpecka) [inline cutscene_engine/DialogueBox.gd:6]: Dead code

File: cutscene_engine/DialogueBox.gd [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		queue_free()
+	dialogue_index += 1
+	
+func test_modulate():
+	var stroheim_pfp = $stroheim_profile_ph
+	var stroheim_tween = $Tween_profile
+	var current_color = stroheim_pfp.self_modulate
+	if Input.is_action_just_pressed("fire"):
+		stroheim_tween.interpolate_property(stroheim_pfp, "self_modulate", current_color, Color8(1,1,1,0),0.5,Tween.TRANS_LINEAR,Tween.EASE_OUT)
+		#stroheim_tween.interpolate_property(stroheim_pfp, "self_modulate", Color8(1,1,1,0), current_color,0.5,Tween.TRANS_LINEAR,Tween.EASE_OUT)
```
Comment (rpecka) [inline cutscene_engine/DialogueBox.gd:48]: Dead code

**Review by rpecka — CHANGES_REQUESTED** [review]
Looks like a good start. Since we are a little spread out, it would be nice if we didn't have unused files/code laying around so since you aren't using them yet, can you remove or reformat the json files until the system we talked about is in place?

**Comment by jfdoming** [comment]
Any status update on this?

**Comment by rpecka** [comment]
> Any status update on this?

Hey, sorry for the late reply. @Telvy is working on a followup to this at the moment. I'm not sure how much of what he is working on now depends on this code but he is new to git so I think the second PR will include everything done in this. We should probably close this.

@Telvy did you use the code in this PR in the new stuff you're working on now? Have you made significant changes to the code in this PR since you submitted it?

**Comment by Telvy** [comment]
Heyya. Sorry for the late reply. For the past week or so I've been creating a completely new system that has more classes implemented to have the whole system working.

The whole system reads a .json file with information on the respective characters, expressions and lines. Most the code in my first pull request is practically dead at this point. I'll scrap the code for the visual text scrolling affects and that's it.

Also first time comment replying on github. Never did this before but I hope I'm doing this right.

**Comment by jfdoming** [comment]
@Telvy all good, sounds exciting! Let's close this PR then in favour of the new one.

**Pull request closed without merging**